cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -322,39 +320,62 @@ def __init__(self, app,
         logger.info("constructed %s:%s for data_server_call_back=%s", self.__class__, self, data_server_call_back)
 
     @staticmethod
-    def inspect_state(app):
+    def free_up_space(app):
         token = request.args.get('token', None)
-        if token is None:
-            # TODO what about using the RequestNotAuthorized exception?
-            # raise RequestNotAuthorized("A token must be provided.")
-            return make_response('A token must be provided.'), 403
-        try:
-            secret_key = app.config.get('conf').secret_key
-            decoded_token = tokenHelper.get_decoded_token(token, secret_key)
-            logger.info("==> token %s", decoded_token)
-        except jwt.exceptions.ExpiredSignatureError:
-            # raise RequestNotAuthorized("The token provided is expired.")
-            return make_response('The token provided is expired.'), 403
-        except jwt.exceptions.InvalidTokenError:
-            # raise RequestNotAuthorized("The token provided is not valid.")
-            return make_response('The token provided is not valid.'), 403
 
-        recent_days = request.args.get('recent_days', 3, type=float)
-        job_id = request.args.get('job_id', None)
+        app_config = app.config.get('conf')
+        secret_key = app_config.secret_key
 
-        roles = tokenHelper.get_token_roles(decoded_token)
+        output, output_code = tokenHelper.validate_token_from_request(token=token, secret_key=secret_key,
```

Comment:
there should be a special role for this, like mmoda-ops

Reply:
I was planning on using the already existing role `job-manager` already used for the inspect state endpoint, why do you think another one, specific for this, is necessary? We are having more and more roles, which many of which are specific for single operations.

Reply:
The roles first of all are meant to group people, nothing wrong with them being specific to operations if the operations are meaningfully distinct.
Here, removing directories is distinct in that it is invasive. In fact, nobody should be allowed to do this except for the bot.